mobile/src/test/java/org/openhab/habdroid/model/OpenHABWidgetTest.java

```diff
@@ -0,0 +1,418 @@
+    List<OpenHABWidget> sut2 = new ArrayList<>();
+    List<OpenHABWidget> sut3 = new ArrayList<>();
+
+    @Test
+    @Before
+    public void parse_createsOpenHABWidget() throws Exception {
+        OpenHABWidget.parseXml(sutXml, null, createXmlNode());
+        OpenHABWidget.parseJson(sut1, null, createJSONObject(1), "PNG");
+        OpenHABWidget.parseJson(sut2, null, createJSONObject(2), "SVG");
+        OpenHABWidget.parseJson(sut3, null, createJSONObject(3), "SVG");
+    }
+
+    @Test
+    public void testCountInstances() throws Exception {
+        assertEquals(sutXml.size(), 2);
+        assertEquals(sut1.size(), 2);
+        assertEquals(sut2.size(), 1);
+        assertEquals(sut3.size(), 4);
+    }
+
+    @Test
+    public void getIconPath_iconExists_returnIconUrlfromImages() throws Exception {
+        assertEquals("images/groupicon.png", sutXml.get(0).iconPath());
+    }
+
+    @Test
+    public void testGetIconPath() throws Exception {
+        assertEquals("icon/groupicon?state=OFF&format=PNG", sut1.get(0).iconPath());
+        assertEquals("icon/groupicon?state=ON&format=SVG", sut2.get(0).iconPath());
+        assertEquals("icon/slider?state=81&format=SVG", sut3.get(1).iconPath());
+        assertEquals("Rollersutter icon should always be 0 to 100, not ON/OFF", "icon/rollershutter?state=0&format=SVG", sut3.get(2).iconPath());
+        assertEquals("icon/rollershutter?state=42&format=SVG", sut3.get(3).iconPath());
+    }
+
+    @Test
+    public void testGetChildren() throws Exception {
+        assertEquals("demo11", sut1.get(1).id());
+        assertEquals("0202_0_0_1", sut3.get(2).id());
+    }
+
+    @Test
+    public void testGetPeriod() throws Exception {
+        assertEquals("M", sut1.get(0).period());
+        assertEquals("Object has no period, should default to D", "D", sut2.get(0).period());
+    }
+
+    @Test
+    public void testGetStep() throws Exception {
+        assertEquals(1F, sut1.get(0).step());
+        assertEquals(0.1F, sut2.get(0).step());
+    }
+
+    @Test
+    public void testGetRefresh() throws Exception {
+        assertEquals(1000, sut1.get(0).refresh());
+        assertEquals("Min refresh is 100, object has set refresh to 10",100, sut2.get(0).refresh());
+    }
+
+    @Test
+    public void testHasItem() throws Exception {
+        assertNotNull(sut1.get(0).item());
+        assertNotNull(sut2.get(0).item());
+    }
+
+    @Test
+    public void testHasLinkedPage() throws Exception {
+        assertNotNull(sut1.get(0).linkedPage());
+        assertNull(sut2.get(0).linkedPage());
+    }
+
+    @Test
+    public void testHasMappings() throws Exception {
+        assertEquals(true, sut1.get(0).hasMappings());
+        assertEquals(true, sut2.get(0).hasMappings());
+    }
+
+    @Test
+    public void testGetColors() throws Exception {
+        Assume.assumeThat(Color.parseColor("blue"), equalTo(Color.BLUE));
+        // Doesn't seem to work via Android Studio
```

Comment:
I haven't tested it, when I added this test, but doesn't it work, when you move it to mobile/src/androidTest/java/org/openhab/habdroid/model/OpenHABWidgetTest.java ?

Reply:
Couldn't we also use the constant (#ff0000ff that is)? Moving this just for the constant definition seems a bit too much.

Reply:
Removed the assumption, it was unneeded anyway (blue wasn't used in test JSON).